misc/codegen/lib/ql.py

```diff
@@ -115,6 +116,8 @@ class Class:
 
     def __post_init__(self):
         self.bases = [Base(str(b), str(prev)) for b, prev in zip(self.bases, itertools.chain([""], self.bases))]
+        self.bases_impl = [Base(str(b), str(prev)) for b, prev in zip(
+            self.bases_impl, itertools.chain([""], self.bases_impl))]
```

Comment:
nit, there's some duplication here. Maybe
```suggestion
        get_bases = lambda bases: [Base(str(b), str(prev)) for b, prev in zip(bases, itertools.chain([""], bases))]
        self.bases = get_bases(self.bases)
        self.bases_impl = get_bases(self.bases_impl)
```
?